framework/web/AssetManager.php

```diff
@@ -572,10 +594,15 @@ public function getPublishedUrl($path)
      * Generate a CRC32 hash for the directory path. Collisions are higher
      * than MD5 but generates a much smaller hash string.
      * @param string $path string to be hashed.
+     * @param boolean $isFile whether the path is file.
      * @return string hashed string.
      */
-    protected function hash($path)
+    protected function hash($path, $isFile = true)
```

Comment:
Agree, at first I was going to do make

```
    protected function hash($path)
    {
        if (!empty($this->hashCallback)) {
            return call_user_func($this->hashCallback, $path, $isFile);
        }
        $path = (is_file($path) ? dirname($path) : $path) . filemtime($path);
        return sprintf('%x', crc32($path . Yii::getVersion()));
    }
```

But then, there will be a redundant `is_file($path)` calls. If you want to trade that off with function signature, I'm totally okay.